clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -258,7 +258,10 @@ public void poll(long now) {
         }
 
         pollHeartbeat(now);
-        maybeAutoCommitOffsetsAsync(now);
+        if (subscriptions.hasNoSubscription())
+            subscriptions.clearAssignment();
```

Comment:
That's true, but perhaps that could be a call to `clearAssignment` directly from `unsubscribe`? If we only clear the assignment here, then it wouldn't be cleared until the user calls `poll` after calling `unsubscribe` which doesn't seem quite right.